src/tcgwars/logic/impl/gen3/LegendMaker.groovy

```diff
@@ -1512,10 +1528,11 @@ public enum LegendMaker implements LogicCardInfo {
         move "Dream Eater", {
           text "30 damage. If the Defending Pokémon is not Asleep, this attack does nothing."
           energyCost P
+          attackRequirement{
+            assert defending.isSPC(ASLEEP) : "The Defending Pokémon is not asleep"
+          }
           onAttack {
-            if (defending.isSPC(ASLEEP)) {
-              damage 30
-            }
+            damage 30
```

Comment:
The if statement is still required for copying this attack AFAIK.

Reply:
Ok, I may have been removing a couple of these if thinking of them as unneeded. Will avoid doing so in the future, thanks for the correction 👌